### PR TITLE
Adding support from Koa to Olive versions in legacy mode

### PR DESCRIPTION
**Pull request by Ian2012**
This PR has the intention to make some changes that will allow users of the seb browser plugin to install it in versions later than hawthorn, so far the correct operation from Koa to Olive version in legacy mode has been tested, to achieve this goal the eduNEXT team made the following modifications:

- We removed the previous CI that was in CircleCI and added a new set using Github Actions, this test set was implemented to test unit tests in both Python 3.6 and 3.8.
-  Python 2.7 support was removed and only python 3 support was left.
- New backends were added for nutmeg, these backends were tested and work correctly for versions from Koa to Olive.
- Updated the version of the unit test dependencies to match those used by the platform in nutmeg.

### How to run unit tests locally

1. To run the unit tests, a Python virtual environment must be created.
`mkvirtualenv venv`
2. Activate your virtual environment
`source venv/bin/activate`
3. Install unit test requirements.
`pip install -r requirements.txt`
4. Run the tests.
`coverage run --source="." -m pytest`

### How to test the plugin functionalities manually

1. Create a yaml tutor plugin and activate it, to know further details about ysml tutor plugins, read the following article [YAML file](https://docs.tutor.overhang.io/plugins/v0/gettingstarted.html#yaml-file).
``` yaml
name: seb-backend-plugin
version: 0.1.0
patches:
  openedx-common-settings: |
    MIDDLEWARE.append("seb_openedx.middleware.SecureExamBrowserMiddleware")

  common-env-features: |
    ENABLE_OTHER_COURSE_SETTINGS: true
```
2. Install the seb-openedx plugin as an extra requirement for more information on how to install an extra requirement in tutor you should follow the article below [Installing extra xblocks and requirements](https://docs.tutor.overhang.io/configuration.html#installing-extra-xblocks-and-requirements).
3. Configure the plugin, in the following link there is useful information for its correct configuration [seb-openedx documentation](https://seb-openedx.readthedocs.io/).

### CODE REVIEW

**Comment by Ian2012** [comment]
@felipemontoya it's not ready until we confirm in which version this variable was changed from `COURSE_OUTLINE_PAGE_FLAG` to `DISABLE_COURSE_OUTLINE_PAGE_FLAG`

**Comment by DeimerM** [comment]
> @felipemontoya it's not ready until we confirm in which version this variable was changed from `COURSE_OUTLINE_PAGE_FLAG` to `DISABLE_COURSE_OUTLINE_PAGE_FLAG`

 this variable was deprecated since koa so it's compatible with all versions we are testing, you can find the commit in which was replaced in the following link https://github.com/openedx/edx-platform/commit/13a70fcaa64428fd564e912aa370a96ce9ee3316

**Comment by felipemontoya** [comment]
The commit you link was first available for the koa.1 release. It was not in master before the koa cut but was literally made during the koa process.

**Review by felipemontoya — APPROVED** [review]
The whole set of changes is what I would expect.
We also tested this branch in the integration server.

@Ian2012 after we merge this we should inmediately tag it with version v2.0.0

**Comment by felipemontoya** [comment]
I would suggest we merge this after we have added the changelog file.

File: .github/workflows/ci.yml
Comment (felipemontoya) [inline .github/workflows/ci.yml:45]: I know noticed this --exit-zero flag.

Is there a reason for this? shouldn't we run pylint normally and use exceptions where the errors are?

Comment (Ian2012) [reply .github/workflows/ci.yml:45]: This flag allows github actions to know pylint has finished gracefully, we can also use pylint-exit:

https://stackoverflow.com/questions/71227177/github-action-error-process-completed-with-exit-code-14

Comment (felipemontoya) [reply .github/workflows/ci.yml:45]: Reading the flag explanation it should definitely stay, but I wonder then why the Pyling analysis finished with 9.91 rate and it did not fail.

To that end, I think we should add 	`duplicate-code` to the list of disabled messages in `.pylintrc`. Then we would be left with 7 warning that we can ignore one by one.

Comment (Ian2012) [reply .github/workflows/ci.yml:45]: @felipemontoya ready

**Comment by felipemontoya** [comment]
We are good to merge then

**Pull request merged**
